libs/coin-framework/src/currencies/support.ts

```diff
@@ -63,15 +61,62 @@ setSupportedFiats([
   "VND",
   "VUV",
   "ZAR",
-]);
+];
 
-export function isFiatSupported(fiat: FiatCurrency) {
-  return userSupportedFiats.includes(fiat);
+async function initializeUserSupportedFiats() {
+  try {
+    const ids = await fetchSupportedFiatsTokens();
+    const idsToUpper = ids.map(id => id.toUpperCase());
+
+    // This makes sure we only keep the elements supported in our API and that are available for getFiatCurrencyByTicker
+    const supportedTokens = idsToUpper.filter(token => locallySupportedFiats.includes(token));
+    userSupportedFiats = supportedTokens.map(id => {
+      return getFiatCurrencyByTicker(id);
+    });
+  } catch (error) {
+    console.error("Error:", error);
+  }
+}
+
+export async function fetchSupportedFiatsTokens(): Promise<string[]> {
+  try {
+    const response = await fetch("https://countervalues.live.ledger.com/v2/supported-to", {
+      method: "GET",
+      headers: {
+        accept: "application/json",
+      },
+    });
+
+    if (!response.ok) {
+      throw new Error(`Request failed with status ${response.status}`);
+    }
+
+    const data: string[] = await response.json();
+    return data;
+  } catch (error) {
+    // Handle any network or parsing errors here
+    console.error("Error:", error);
+    throw error;
+  }
 }
-export function setSupportedFiats(ids: string[]) {
-  userSupportedFiats = ids.map(getFiatCurrencyByTicker);
+
+// Usage of isFiatSupported and listSupportedFiats should check if userSupportedFiats is populated
+export function isFiatSupported(fiat: FiatCurrency) {
+  return userSupportedFiats?.includes(fiat);
 }
-export function listSupportedFiats(): FiatCurrency[] {
+
+export async function listSupportedFiats(): Promise<FiatCurrency[]> {
+  if (userSupportedFiats === null) {
+    // Handle case where userSupportedFiats is not yet populated (e.g., by calling initializeUserSupportedFiats)
+    try {
+      await initializeUserSupportedFiats();
+    } catch (error) {
+      // Handle initialization error
+      console.error("Failed to initialize userSupportedFiats:", error);
```

Comment:
this is not really tracked like this. we should use `log` so it's included in exported logs at least.